slice/internal/controller/workload_controller.go

```diff
@@ -413,36 +504,94 @@ func (r *WorkloadReconciler) updateWorkloadAdmissionCheckStatus(ctx context.Cont
 	originalState := ac.State
+	originalMessage := ac.Message
 
-	errCond := meta.FindStatusCondition(slice.Status.Conditions, string(v1alpha1.Error))
+	prepareAdmissionCheckStatus(ac, slices)
 
-	switch {
-	case core.Forming(slice):
-		ac.Message = fmt.Sprintf("The Slice %q is being formed", slice.Name)
-	case core.Ready(slice):
-		ac.State = kueue.CheckStateReady
-		ac.Message = fmt.Sprintf("The Slice %q is fully operational", slice.Name)
-	case core.Degraded(slice):
-		ac.State = kueue.CheckStateReady
-		ac.Message = fmt.Sprintf("The Slice %q is running with reduced capacity or performance", slice.Name)
-	case core.Deformed(slice):
-		ac.State = kueue.CheckStateRejected
-		ac.Message = fmt.Sprintf("The Slice %q is being torn down", slice.Name)
-	case errCond != nil && errCond.Status == metav1.ConditionTrue:
-		ac.State = kueue.CheckStateRejected
-		ac.Message = fmt.Sprintf("The Slice %q is not operational due to an error: %s", slice.Name, errCond.Message)
+	// No changes.
+	if originalState == ac.State && ac.Message == originalMessage {
+		return nil
 	}
 
-	err := r.updateWorkloadAdmissionCheckStatus(ctx, wl, ac)
-	if err == nil && originalState != ac.State {
+	if err := r.updateWorkloadAdmissionCheckStatus(ctx, wl, ac); err != nil {
+		return err
+	}
+
+	log := ctrl.LoggerFrom(ctx)
+
+	if originalState != ac.State {
 		message := fmt.Sprintf("Admission check %q updated state from %q to %q", ac.Name, originalState, ac.State)
+		log.V(3).Info(message)
 		r.record.Event(wl, corev1.EventTypeNormal, AdmissionCheckUpdatedEventType, message)
 	}
 
-	return err
+	if ac.Message != originalMessage {
+		// Logging error messages if exists
+		for _, slice := range slices {
+			cond := meta.FindStatusCondition(slice.Status.Conditions, string(v1alpha1.Error))
+			if cond != nil {
+				log.V(2).Info(
+					"WARNING: The Slice are not operational due to an error",
+					"slice", klog.KObj(&slice),
+					"error", cond.Message,
+				)
+			}
+		}
+	}
+
+	return nil
+}
+
+func groupSlicesByStatus(slices []v1alpha1.Slice) map[v1alpha1.SliceConditionType][]v1alpha1.Slice {
+	slicesByStatus := make(map[v1alpha1.SliceConditionType][]v1alpha1.Slice)
+	for _, slice := range slices {
+		for _, status := range core.SliceStates {
+			if meta.IsStatusConditionTrue(slice.Status.Conditions, string(status)) {
+				slicesByStatus[status] = append(slicesByStatus[status], slice)
+				break
+			}
+		}
+	}
+	return slicesByStatus
+}
+
+func prepareAdmissionCheckStatus(ac *kueue.AdmissionCheckState, slices []v1alpha1.Slice) {
+	slicesByStatus := groupSlicesByStatus(slices)
+
+	switch {
+	case len(slicesByStatus[v1alpha1.Error]) > 0 || len(slicesByStatus[v1alpha1.Deformed]) > 0:
+		ac.State = kueue.CheckStateRejected
+	case len(slices) == len(slicesByStatus[v1alpha1.Degraded])+len(slicesByStatus[v1alpha1.Ready]):
+		ac.State = kueue.CheckStateReady
+	}
+
+	ac.Message = fmt.Sprintf("Slices are in states: %d Created", len(slices))
```

Comment:
are we double counting? so it could be 3 Created, 3 Ready, when in total there are just 3?

Reply:
Fixed.